core/dbt/contracts/graph/nodes.py

```diff
@@ -608,32 +654,99 @@ def same_contract(self, old) -> bool:
         contract_enforced_disabled: bool = False
         columns_removed: List[str] = []
         column_type_changes: List[Tuple[str, str, str]] = []
+        enforced_column_constraint_removed: List[List[str]] = []  # column, constraint_type
```

Comment:
any reason not to use a list of `Tuple[str, str]` here and for `enforced_model_constraint_removed`? Feels more consistent with the implementation so far (of `columns_removed`) given the nested lists are of fixed length.